File: ai-hub-backend/src/models/taskModel.js
import mongoose from 'mongoose';

const taskSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true
  },
  description: String,
  projectId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Project',
    required: true
  },
  apiMethod: {
    type: String,
    required: true
  },
  status: {
    type: String,
    enum: ['archived', 'active', 'inactive'],
    default: 'active'
  },
  prompts: [{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Prompt'
  }],
  createdBy: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Admin',
    required: true
  }
}, { timestamps: true });

export default mongoose.model('Task', taskSchema);

File: backend/infra/db/mongoose/models/task_model.go
package models

import "time"

type TaskDoc struct {
	ID          string    `bson:"_id"`
	Name        string    `bson:"name"`
	Description *string   `bson:"description,omitempty"`
	ProjectId   string    `bson:"projectId"`
	APIMethod   string    `bson:"apiMethod"`
	Status      string    `bson:"status"`
	CreatedBy   string    `bson:"createdBy"`
	CreatedAt   time.Time `bson:"createdAt"`
	UpdatedAt   time.Time `bson:"updatedAt"`
}
